xarray/core/dtypes.py

```diff
@@ -40,7 +64,7 @@ def maybe_promote(dtype):
     return np.dtype(dtype), fill_value
 
 
-def get_fill_value(dtype):
+def get_fill_value(dtype, fill_value_typ=None):
```

Comment:
Can we make separate functions for this, maybe `get_pos_infinity` and `get_neg_infinity`? It feels a little strange to put it all in one function, and with separate functions you can avoid the need to validate the `fill_value_typ` argument.